rpc/src/client/transport/http.rs

```diff
@@ -196,7 +222,7 @@ impl Client for HttpClient {
                 // Back-fill with a request to /block endpoint and
                 // taking just the header from the response.
                 let resp = self
-                    .perform_v0_34(endpoint::block::Request::new(height))
+                    .execute::<_, v0_34::Dialect>(endpoint::block::Request::new(height))
```

Comment:
I am not a huge fan of functions which won't compile without explicitly passing them type argument, how about passing the dialect as an value-level argument instead?

ie. change the definition of `execute` to:

```rust
    async fn execute<R, S>(&self, request: R, dialect: S) -> Result<R::Output, Error>
    where
        R: SimpleRequest<S>,
        S: Dialect,
    {
```

What do you think?

Reply:
Yes, stylistically it's better.

Reply:
Done in https://github.com/informalsystems/tendermint-rs/pull/1362/commits/8cc408d5770c41c75bd12da403093ae610ce4340 and https://github.com/informalsystems/tendermint-rs/pull/1362/commits/a41c4abe39cfa9a86d9d7081071a03bf56666894